server/lib/pbench/indexer.py

```diff
@@ -3524,36 +3979,80 @@ def __init__(self, idxctx, tbarg, tmpdir, extracted_root):
         # It's what ties all of the relevant documents together.
         self.run_metadata["id"] = md5sum
 
+    def gen_files_by_partial_path(self, path):
+        """Generator for all files in the tar ball which match the given path
+        pattern.
+        """
+        for member in self.members:
+            if member.isfile() and member.name.find(path) >= 0:
```

Comment:
This looks fine, but can I suggest we use pathlib here? https://docs.python.org/3/library/pathlib.html

Reply:
Seems like we don't gain much but syntactic changes.  Am I missing something?

Reply:
Let's track this in issue #1584 and make this a deliberate change in a separate commit.